### ISSUE TITLE
Problems with search and themes when refreshing

### ISSUE THREAD

**Issue by NullDott** [Laeoying/pinterestdarktheme#3]
## Search
There's a rendering issue in the search bar on Pinterest when the extension is enabled, causing it to be transparent, seems to happen when the below issue occurs too, or the white theme is active:
![image](https://github.com/Laeoying/pinterestdarktheme/assets/101472580/5c43238d-3167-48c7-a573-6c6b6bc77a6f)

## Themes
There's also a problem when refreshing Pinterest where the applied theme will reset to white, even when the "Always allow in domain" option is active, this makes the extension practically unusable since you have to apply the theme over and over again when refreshing the [page.]

https://github.com/Laeoying/pinterestdarktheme/assets/101472580/4117dfc2-a101-4e23-8409-e40ab9339063

**Firefox 126**

**Comment by Laeoying** [comment]
Yup I saw that when I've posted it on Mozilla. I'll try to fix it when I'll have time, don't have much right now :/
Thanks for reporting tho :)

**Comment by Laeoying** [comment]
Btw the transparent layer append when there's no enabled theme I believe, eg when you load the page and didn't select any theme.
Actually, the extension is using Chrome storage, that's why it doesn't work I think.

**Comment by NullDott** [comment]
Thanks so much!

**Comment by NullDott** [comment]
The issue seems to keep happening after the last update, is it related to the `manifest.json` mentioned?

**Comment by Filipermaster** [comment]
After some proper research, I found out the issue was due to minor differences between Chrome's and Mozilla's storage APIs, making the lazy fix useless. Those cases, and some others, were hopefully accounted for, so expect the next update to at least render the plugin usable. The only compromise might be the color fade effect not work on Firefox.

**Comment by Laeoying** [comment]
I nearly finished my exams so I'll work on it soon :)

**Comment by NullDott** [comment]
It's okay! Take your time 😊

**Comment by Laeoying** [comment]
I'm closing this since I think it's fixed (at least I hadn't this problem in a while, maybe there are some settings to toggle in Firefox), if the issue reappear I'll make another issue.
